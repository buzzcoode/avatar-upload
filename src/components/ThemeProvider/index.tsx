import React from 'react'
import { ThemeProvider as EmotionThemeProvider } from 'emotion-theming';
import { theme } from '../../utils/theme';

interface ThemeProviderProps {
  children: React.ReactNode
}

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  return (
    <EmotionThemeProvider theme={theme}>
      {children}
    </EmotionThemeProvider>
  )
}
