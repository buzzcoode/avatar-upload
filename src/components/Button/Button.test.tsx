import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { Button } from '.';
import { ThemeProvider } from '../ThemeProvider';

it('renders correctly', async () => {
  const { queryByTestId } = render(
    <ThemeProvider>
      <Button>content</Button>
    </ThemeProvider>
  );
  expect(queryByTestId('styled-button')).toMatchSnapshot();
});

it('renders snapshot', async () => {
  const { container } = render(
    <ThemeProvider>
      <Button textColor="red">click me</Button>
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

it('triggers clicks correctly', async () => {
  const onClick = jest.fn()
  const { getByTestId } = render(
    <ThemeProvider>
      <Button onClick={onClick}>click me</Button>
    </ThemeProvider>
  );
  fireEvent.click(getByTestId('styled-button'))
  expect(onClick).toBeCalledTimes(1);
});
