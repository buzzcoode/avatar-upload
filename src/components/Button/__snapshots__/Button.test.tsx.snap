// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-0 {
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  background-color: #3F80FE;
  color: #3F80FE;
  background: none;
}

<button
  class="emotion-0"
  data-testid="styled-button"
>
  content
</button>
`;

exports[`renders snapshot 1`] = `
.emotion-0 {
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  background-color: #3F80FE;
  color: red;
  background: none;
}

<div>
  <button
    class="emotion-0"
    data-testid="styled-button"
  >
    click me
  </button>
</div>
`;
